### PR TITLE
Allows user to enter country code into phone number authentication field (E.g. by copy/pasting or from keyboard suggestions)

### PR DESCRIPTION
**Pull request by andyboyd**
This references issue #529

### CODE REVIEW

**Comment by andyboyd** [comment]
I signed the CLA...

**Comment by andyboyd** [comment]
Signed the CLA with my other email address as well...

File: FirebasePhoneAuthUI/FUIPhoneEntryViewController.m [hunk 1]
```diff
@@ -187,8 +187,9 @@ - (void)onNext:(NSString *)phoneNumber {
   [self incrementActivity];
   self.navigationItem.rightBarButtonItem.enabled = NO;
   FIRPhoneAuthProvider *provider = [FIRPhoneAuthProvider providerWithAuth:self.auth];
-  NSString *phoneNumberWithCountryCode =
-      [NSString stringWithFormat:@"+%@%@", _selectedCountryCode.dialCode, phoneNumber];
+  NSString *selectedCountryCodeString = [NSString stringWithFormat:@"+%@", _selectedCountryCode.dialCode ];
+  BOOL isPhoneNumberAlreadyPrefixed = [phoneNumber hasPrefix:selectedCountryCodeString];
+  NSString *phoneNumberWithCountryCode = (isPhoneNumberAlreadyPrefixed) ? phoneNumber : [NSString stringWithFormat:@"%@%@", selectedCountryCodeString, phoneNumber];
```
Comment (Yue-Wang-Google) [inline FirebasePhoneAuthUI/FUIPhoneEntryViewController.m]: we have a 100-char-per-line limit. so you can write this as:

```
NSString *selectedCountryCodeString = [NSString stringWithFormat:@"+%@", _selectedCountryCode.dialCode];
BOOL isPhoneNumberAlreadyPrefixed = [phoneNumber hasPrefix:selectedCountryCodeString];
NSString *phoneNumberWithCountryCode;
if (isPhoneNumberAlreadyPrefixed) {
  phoneNumberWithCountryCode = phoneNumber;
} else {
  phoneNumberWithCountryCode =
      [NSString stringWithFormat:@"%@%@", selectedCountryCodeString, phoneNumber];
}
```

File: FirebasePhoneAuthUI/FUIPhoneEntryViewController.m [hunk 2]
```diff
@@ -187,8 +187,9 @@ - (void)onNext:(NSString *)phoneNumber {
   [self incrementActivity];
   self.navigationItem.rightBarButtonItem.enabled = NO;
   FIRPhoneAuthProvider *provider = [FIRPhoneAuthProvider providerWithAuth:self.auth];
-  NSString *phoneNumberWithCountryCode =
-      [NSString stringWithFormat:@"+%@%@", _selectedCountryCode.dialCode, phoneNumber];
+  NSString *selectedCountryCodeString = [NSString stringWithFormat:@"+%@", _selectedCountryCode.dialCode ];
```
Comment (Yue-Wang-Google) [inline FirebasePhoneAuthUI/FUIPhoneEntryViewController.m]: extra space before `]`

**Comment by andyboyd** [comment]
Updated with the requested changes from @Yue-Wang-Google

**Comment by Yue-Wang-Google** [comment]
PR merged. Thanks andyboyd@ for doing this!

**Pull request merged**